re_pages/Packages.res

```diff
@@ -516,6 +527,16 @@ let default = (props: props) => {
     </Category>
   }
 
+  // Necessary, because the router query is undefined on initial render.
```

Comment:
Maybe take some inspiration on how I did it on the [Syntax Widget PR](https://github.com/reason-association/rescript-lang.org/pull/114/files#diff-00b54ebd4efe3d1fb5cbb8ac52f08bd216da3f7baa00f57e7891831679b20040R252).

I use `useEffect0` to trigger the first state transition based on the urlpath. Later on I use `useEffect1` based on the state to trigger the url replacing

Here it feels like it's kinda the other way round (reacting on `router.query` changes, instead of state changes), which feels wrong to me.

Reply:
So the general idea should IMO be:
- On initial render, check if there's a `search=...` query param, set the initial state based on that in `useEffect0`
- Whenever the state changes to some new `Filter(search)`, run a side-effect to trigger `scrollToTop`, and replace the url search term (`useEffect1` based on the `Packages` component `state`)

Reply:
I guess it is now more "the other way around", but only with one more Ref and Effect.
Not sure if that is better than the one hook with the router.query as a dependency. It certainly triggers only twice (initial render and then the second render where the `query` is accessible if available).